components/camel-elasticsearch5-rest/src/main/java/org/apache/camel/component/elasticsearch5/ElasticsearchProducer.java

```diff
@@ -0,0 +1,295 @@
+            searchRequest.source(sourceBuilder);
+            try {
+                restHighLevelClient.search(searchRequest);
+                message.setBody(true);
+            } catch (ElasticsearchStatusException e) {
+                if (e.status().equals(RestStatus.NOT_FOUND)) {
+                    message.setBody(false);
+                } else {
+                    throw new IllegalStateException(e);
+                }
+
+            }
+        } else if (operation == ElasticsearchOperation.SEARCH) {
+            SearchRequest searchRequest = message.getBody(SearchRequest.class);
+            message.setBody(restHighLevelClient.search(searchRequest).getHits());
+        } else {
+            throw new IllegalArgumentException(ElasticsearchConstants.PARAM_OPERATION + " value '" + operation + "' is not supported");
+        }
+
+        // If we set params via the configuration on this exchange, remove them
+        // now. This preserves legacy behavior for this component and enables a
+        // use case where one message can be sent to multiple elasticsearch
+        // endpoints where the user is relying on the endpoint configuration
+        // (index/type) rather than header values. If we do not clear this out
+        // sending the same message (index request, for example) to multiple
+        // elasticsearch endpoints would have the effect overriding any
+        // subsequent endpoint index/type with the first endpoint index/type.
+        if (configIndexName) {
+            message.removeHeader(ElasticsearchConstants.PARAM_INDEX_NAME);
+        }
+
+        if (configIndexType) {
+            message.removeHeader(ElasticsearchConstants.PARAM_INDEX_TYPE);
+        }
+
+        if (configWaitForActiveShards) {
+            message.removeHeader(ElasticsearchConstants.PARAM_WAIT_FOR_ACTIVE_SHARDS);
+        }
+        if (configuration.getDisconnect()) {
+            client.close();
+            client = null;
+            if (configuration.getEnableSniffer()) {
+                sniffer.close();
+                sniffer = null;
+            }
+        }
+
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    protected void doStart() throws Exception {
+        super.doStart();
+        if (!configuration.getDisconnect()) {
+            startClient();
+        }
+    }
+
+    private void startClient() throws NoSuchMethodException, IllegalAccessException, InvocationTargetException, InstantiationException, UnknownHostException {
+        if (client == null) {
+            LOG.info("Connecting to the ElasticSearch cluster: " + configuration.getClusterName());
+            if (configuration.getHostAddressesList() != null
+                && !configuration.getHostAddressesList().isEmpty()) {
+                client = createClient();
+            } else {
+                LOG.info("Incorrect ip address and port parameters settings for ElasticSearch cluster");
```

Comment:
Should this not be WARN or some exception thrown? Instead of logging that the ip is incorrect. And maybe this validation should be done in the doStart method so its validated earlier